courses/C/intermediate-level/Error-handling/types.md

```diff
@@ -0,0 +1,132 @@
+---
+id: lesson-2
+title: "Error Handling in C"
+sidebar_label:  Error Handling
+sidebar_position: 2
+description: "Learn Error Handling in C"
+tags: [courses,intermediate-level,File Handling,Introduction]
+--- 
+ 
+
+Error handling is an essential aspect of programming that helps manage and respond to issues that arise during the execution of a program. In C, errors can be categorized into several types, and there are specific techniques and tools for handling them.
+
+#### Types of Errors
+
+1. **Syntax Errors**
+   - **Definition**: Errors that occur due to incorrect use of the C language syntax. These are typically caught by the compiler.
+   - **Examples**:
+     ```c
+     int main() {
+         int x = 5
+         return 0;
+     }
+     ```
+     *Missing semicolon after `int x = 5`.*
+
+2. **Runtime Errors**
+   - **Definition**: Errors that occur during the execution of the program. These errors are not detected during compilation.
+   - **Examples**:
+     ```c
+     int main() {
+         int *p = NULL;
+         *p = 10; // Dereferencing a null pointer
+         return 0;
+     }
+     ```
+     *Dereferencing a null pointer.*
+
+3. **Logical Errors**
+   - **Definition**: Errors that occur when the program runs but produces incorrect results due to flawed logic.
+   - **Examples**:
+     ```c
+     int main() {
+         int a = 5, b = 10;
+         int sum = a - b; // Incorrect operation
+         printf("Sum: %d\n", sum);
+         return 0;
+     }
+     ```
+     *Incorrect calculation of sum.*
+
+
+#### Flowchart for Error Handling
+:::note
+1. **Start**
+2. **Perform Operation (e.g., file operation)**
+3. **Check Return Value**
+   - If the operation fails:
+     - Use `errno` to identify the error.
+     - Use `perror` to print the error message.
+     - Handle the error (e.g., clean up resources, exit program).
+   - If the operation succeeds, proceed with the next step.
+4. **End**
+:::
+
+```
+ +----------------------------+
+ | Start                      |
+ +----------------------------+
+            |
+            v
+ +----------------------------+
+ | Perform Operation          |
+ +----------------------------+
+            |
+            v
+ +----------------------------+
+ | Check Return Value         |
+ +----------------------------+
+          /   \
+       Fail   Success
+        |       |
+        v       v
++------------------+  +------------------------+
+| Use errno to     |  | Continue with the      |
+| identify the     |  | program                |
+| error             |  +------------------------+
++------------------+
+        |
+        v
++----------------------------+
+ | Use perror to print error |
+ | message and handle error  |
+ +----------------------------+
+        |
+        v
+ +----------------------------+
+ | End                        |
+ +----------------------------+
+```
+
```

Comment:
You are working on this pull request to add a new C language course to CodeHarborHub. You need Mermaid diagrams to create the flowchart for file handling, which helps visualize the flow of file operations and error handling. Here are the links to the documentation and an example of how I used Mermaid:

* [Mermaid documentation](https://mermaid.js.org/intro/)
* [Diagram with Mermaid](https://codeharborhub.github.io/community/features#diagrams-with-mermaid)